File: Exercise_3.java
class Solution {
    public int findPeakElement(int[] nums) {
        int s = 0;
        int e = nums.length-1;
        while (s<=e){
            int m = s+(e-s)/2;
            
            boolean mgl = m==0 || nums[m]>nums[m-1];
            boolean mgr = m==nums.length-1 || nums[m]>nums[m+1];
            
            if(mgl && mgr){
                return m;
            }else if(mgl){
                s = m+1;
            } else {
                e = m-1;
            }
        }
    return -1;
    }
}

File: Exercise_3.js
// Time Complexity : O(log n)
// Space Complexity : O(1) 
// Approach : finding the element using binary search, spliting [] / 2 where one of peak element will always be at greater end of the mid

var findPeakElement = function (nums) {
    if (nums == null || nums.length == 0) retunr - 1
    let low = 0
    let high = nums.length - 1
    while (low <= high) {
        const mid = Math.floor(low + (high - low) / 2)
        if ((mid == 0 || nums[mid] > nums[mid - 1]) &&
            (mid == nums.length - 1 || nums[mid] > nums[mid + 1])) {
            return mid
        } else if (mid > 0 && nums[mid] < nums[mid - 1]) {
            high = mid - 1
        } else {
            low = mid + 1
        }
    }
    return -1
};